src/coreclr/gc/gc.cpp

```diff
@@ -22286,13 +22286,29 @@ BOOL gc_heap::gc_mark1 (uint8_t* o)
     return marked;
 }
 
+#ifdef USE_REGIONS
+inline bool in_heap_range (uint8_t* o)
+{
+#ifdef FEATURE_BASICFREEZE
+    // we may have frozen objects in read only segments
+    // outside of the reserved address range of the gc heap
+    return ((g_gc_lowest_address <= o) && (o < g_gc_highest_address));
```

Comment:
Should we also check, in debug mode, that the pointer lies in a read-only segment? That assertion could be helpful to find real bad pointers.

Reply:
Good point - I will add an assert before the return statement:

    assert (((g_gc_lowest_address <= o) && (o < g_gc_highest_address)) || (ro_segment_lookup(o) != nullptr));